src/modules/pages/components/RouteLayouts/UserNavigation.tsx

```diff
@@ -89,31 +120,51 @@ const UserNavigation = () => {
           )}
         </ConnectWalletPopover>
       )}
-      {userCanNavigate && (
-        <GasStationPopover
-          transactionAndMessageGroups={transactionAndMessageGroups}
-        >
-          {({ isOpen, toggle, ref }) => (
-            <>
-              <button
-                type="button"
-                className={
-                  isOpen ? styles.walletAddressActive : styles.walletAddress
-                }
-                ref={ref}
-                onClick={toggle}
-              >
-                <MaskedAddress address={walletAddress} />
-              </button>
-              {readyTransactions >= 1 && (
-                <span className={styles.readyTransactionsCount}>
-                  {readyTransactions}
-                </span>
-              )}
-            </>
-          )}
-        </GasStationPopover>
-      )}
+      <div className={styles.buttonsWrapper}>
+        {userCanNavigate && nativeToken && (
+          <>
+            <button type="button" className={styles.tokens}>
+              <span
+                className={`${styles.dot} ${
+                  (inactiveBalance.gt(0) || totalBalance.isZero()) &&
+                  styles.dotInactive
+                }`}
+              />
+              <Numeral
+                suffix={` ${nativeToken?.symbol} `}
+                unit={getTokenDecimalsWithFallback(nativeToken?.decimals)}
+                value={totalBalance}
+                truncate={3}
+              />
+            </button>
+          </>
+        )}
```

Comment:
This should be moved into a sub-component, along with all the lock / balance checks